src/NuGet.Core/NuGet.CommandLine.XPlat/NuGet.CommandLine.XPlat.csproj

```diff
@@ -20,23 +20,13 @@
   <ItemGroup>
     <PackageReference Include="Microsoft.Extensions.CommandLineUtils.Sources" PrivateAssets="All" />
     <PackageReference Include="System.Diagnostics.Debug" />
+    <PackageReference Include="Microsoft.Build" ExcludeAssets="runtime" />
   </ItemGroup>
 
-  <!-- Microsoft.Build.Locator is needed when debugging, but should not be used in the assemblies we insert.
-       It also generates a build error if the output directory contains MSBuild dlls, so exclude msbuild assets when using the locator. -->
-  <Choose>
-    <When Condition=" '$(Configuration)' == 'Debug' ">
-      <ItemGroup>
-        <PackageReference Include="Microsoft.Build" ExcludeAssets="runtime" />
-        <PackageReference Include="Microsoft.Build.Locator" />
-      </ItemGroup>
-    </When>
-    <Otherwise>
-      <ItemGroup>
-        <PackageReference Include="Microsoft.Build" />
-      </ItemGroup>
-    </Otherwise>
-  </Choose>
+  <!-- Microsoft.Build.Locator is only used when debugging, and the compiler will trim this dependency from Release assemblies we insert. -->
```

Comment:
Note the comment change. I feel this is more clear & explicit about the behavior.

Reply:
Do we need to enable a property to trim the assembly?

Reply:
> Do we need to enable a property to trim the assembly?

You're probably thinking of app trimming, which isonly available on .NET Core as far as I know?
We don't own the end app/don't do publish on our end, so it's only really the SDK that  can do this, but due to the contract that nuget.commandline.xplat and dotnet.exe currently have, that'd be impossible to detect.

Reply:
Even if .NET Core SDK is trimming assemblies, do we trim for .NET Framework?

Reply:
What gets shipped in VS and ilmerge in NuGet.exe is explicitly allow listed with https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.VisualStudio.Client/.vsixinclude and https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.CommandLine/ilmerge.props, so we do trim it, just manually.

Reply:
> I'm not sure the compiler does any trimming here.

It absolutely does! It's easy to see for yourself using ILSpy. Jeff told me about this and I found it pretty impressive that the C# compiler does that.

Reply:
That's not really trimming. 
Trimming involves taking an assembly and pick and choosing only the methods that are used in the app. 

In this case, it just doesn't get copied because you don't compile against it.

Reply:
@nkolev92 could you provide a suggestion for the comment? The only reason the assembly would not be copied is due to it not being referenced based on the DEBUG preprocessor directive. It was not accurate that the Choose/When was actually making a difference.

Reply:
```suggestion
<!-- Microsoft.Build.Locator is only used when debugging, and the compiler will skip copying this dependency from Release assemblies we insert because we only refer to it conditionally with the DEBUG configuration -->
```
>  It was not accurate that the Choose/When was actually making a difference.

It was because the PackageReference was never used on the CI which runs under a Release configuration so that PackageReference was never really added.